internal/api/urls.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+func GetCanonicalApiUrl(userDefinedUrl string) (string, error) {
+	result := ""
+	url, err := url.Parse(userDefinedUrl)
+	if err != nil {
+		return result, err
+	}
+
+	// for localhost we don't change the host, since there are no subdomains
+	if !strings.Contains(url.Host, "localhost") {
```

Comment:
Maybe you could invert the if statement to make it a guard and easier to follow the flow?

Reply:
+1